include/range/v3/action/remove.hpp

```diff
@@ -0,0 +1,82 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Andrey Diduh 2019
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+#ifndef RANGES_V3_ACTION_REMOVE_HPP
+#define RANGES_V3_ACTION_REMOVE_HPP
+
+#include <meta/meta.hpp>
+#include <range/v3/core.hpp>
+#include <range/v3/range_concepts.hpp>
+#include <range/v3/action/action.hpp>
+#include <range/v3/action/remove_if.hpp>
+
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        /// \addtogroup group-actions
+        /// @{
+        namespace action
+        {
+            struct remove_fn
+            {
+            private:
+                friend action_access;
+                template<typename V, typename P = ident, CONCEPT_REQUIRES_(!Range<V>())>
```

Comment:
`!Range<V>()` seems limiting here - I might want to use this action on a range of ranges (e.g. `vector<string>`). Could we split this into two overloads?
```c++
template<typename V>
static auto bind(remove_fn f, V const& v);
template<typename V, typename P,
    CONCEPT_REQUIRES_(!(Range<V>() && EqualityComparable<range_reference_t<V>, P>()))>
static auto bind(remove_fn f, V const& v, P proj);
```

Reply:
```
EqualityComparable<range_reference_t<V>, P>()
```
It never compiles. Probably, compiler proceed processing code even in cases when V is not Range.